### PR TITLE
Allow checking if `MeasurementProcess` objects are in an `Attribute` set.

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
`Attribute.__contains__` raises a `TypeError` if someone tries to use a `MeasurementProcess`. This breaks `qml.measure` as mid-circuit measurements are part of the operator queue, not measurements queue.

**Description of the Change:**
Allow `MeasurementProcess` objects to be used with `Attribute`.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/ops/qubit/attributes.py
Comment (albi3ro) [inline pennylane/ops/qubit/attributes.py:1]: I wonder if it's worth just returning `False` for anything that's not an `Operator`. If the object's not an operator, then it isn't in the attribute. `False`.

**Comment by mudit2812** [comment]
[sc-51773]

**Comment by timmysilv** [comment]
what exactly depends on this module? the change from raising an error to returning False makes sense (if we know _everything_ that should be true, it makes sense to just return false otherwise to me!), but I don't know what effect this will have

**Comment by albi3ro** [comment]
> what exactly depends on this module? the change from raising an error to returning False makes sense (if we know _everything_ that should be true, it makes sense to just return false otherwise to me!), but I don't know what effect this will have

I think basically just a couple of compilation transforms.

**Review by timmysilv — APPROVED** [review]

**Review by albi3ro — APPROVED** [review]

**Pull request merged**